### ISSUE TITLE
Hotfix for log4j 1.2.7 security vulnerability (CVE-2019-17571)

### ISSUE THREAD

**Issue by dongjinleekr** [banzaicloud/koperator#565]
Hello. Kafka is still using log4j 1.2.7, in turn with its [security vulnerability](https://cve.mitre.org/cgi-bin/cvename.cgi?name=CVE-2019-17571). I have been [working on this problem](https://cwiki.apache.org/confluence/display/KAFKA/KIP-653%3A+Upgrade+log4j+to+log4j2) and recently started to provide a [preview patch](https://drive.google.com/file/d/1ar5-vfuip5fSYkh4fN6Pjo2dkzKtafm1/view?usp=sharing) and [custom distribution](https://drive.google.com/file/d/1rdFfQsT9qIkpsaq3sQ3Jokrkef37jLtA/view?usp=sharing) for Apache Kafka 2.7.0.

Some Kafka users consulted me that if they can use this preview with banzaicloud/kafka-operator operator. It seems like banzaicloud/kafka-operator is using some docker image, but I can't guess how to provide a custom docker image for it. Could anyone guide me?

As a note, here is [a examplar custom docker image of it (dongjinleekr/kafka:2.13-2.7.0-log4j2-0)](https://hub.docker.com/r/dongjinleekr/kafka) compatible with [wurstmeister/kafka-docker](https://hub.docker.com/r/wurstmeister/kafka). I hope there is any way to provide similar one.

**Comment by baluchicken** [comment]
Thanks for the heads up. 
A custom docker image can be used by setting the `clusterImage` field in the spec inside the Kafka Cluster CR.
Example:
```
spec:
  headlessServiceEnabled: true
  zkAddresses:
    - "zookeeper-client.zookeeper:2181"
  propagateLabels: false
  oneBrokerPerNode: false
  clusterImage: "dongjinleekr/kafka:2.13-2.7.0-log4j2-0"
  ```

**Comment by dongjinleekr** [comment]
@baluchicken Thanks. I was a great help.

As soon as I publish a migration guide, I will share it here with the instructions for banzaicloud users. :+1:

**Comment by dongjinleekr** [comment]
Here is the guide: https://home.apache.org/~dongjin/post/apache-kafka-log4j2-support/

**Comment by dongjinleekr** [comment]
Long time no see.

Previews & custom patches for [2.6.1, 2.8.0] have been released. (3.0.0 based preview will be released as soon as the official 3.0.0 is released.) Please refer [here](http://home.apache.org/~dongjin/post/apache-kafka-log4j2-support/) to download them.

**Comment by baluchicken** [comment]
Closing this since images are already released.
